File: pe023.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

#define M 28124

bool abundant(int n)
{
    int temp = 1, sq = sqrt(n);
    for (int i = 2; i <= sq; i++) {
        if (n % i == 0) {
            temp += i;
            if (i < n / i)
                temp += n / i;
        }
    }
    return temp > n;
}

int main()
{
    int a[M] = {}, aa = 0;
    for (int i = 1; i < M; i++) {
        if (abundant(i))
            a[aa++] = i;
    }
    
    bool b[M] = {};
    for (int i = 0; i < aa; i++) {
        for (int j = i; j < aa; j++) {
            if (a[i] + a[j] < M)
                b[a[i] + a[j]] = true;
        }
    }
    
    int sum = 0;
    for (int i = 1; i < M; i++)
        if (!b[i])
            sum += i;
    printf("%d\n", sum);
    return 0;
}

File: pe023.py
import math
def abundant(n):
    temp = 1
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            temp += i
            if i != n // i:
                temp += n // i
    return temp > n
m = 28124

a = []
for i in range(1, m):
    if abundant(i):
        a.append(i)

b = [False] * m
for i in range(len(a)):
    for j in range(i, len(a)):
        if a[i] + a[j] < m:
            b[a[i] + a[j]] = True

print(sum(i for i in range(1, m) if not b[i]))
